### PR TITLE
Convert docs to Gitbook format [SAME VERSION]

### PR DESCRIPTION
**Pull request by edrickwong**
**What this PR does / why we need it**:
This PR reformats the `./docs` folder so that it can be parsed by Gitbook 

**Special notes for your reviewer**:

**If applicable**:
- [X] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

**Comment by kate-goldenring** [comment]
Can you clarify the required changes that you made? And possibly point us to the gitbooks documentation that requests this formating? It would be good to have that content as part of the PR history. ALdo, this is a quite a bit to sift through, so it would help me know where to focus. It looks like it is indentation changes, image name changes, and some content/syntax changes.

**Comment by bfjelds** [comment]
it would probably be helpful in the long run to give the images more descriptive names

**Review by kate-goldenring — COMMENTED** [review]
Can you give a before and after tree layout of the docs directory?
Also, can files retain descriptive names?

File: docs/development/building.md [hunk 1]
```diff
@@ -1,36 +1,42 @@
...
 
-In essence, Akri components can be thought of as:
-1. Runtime components
-    1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
-        > Note: For Rust code, build/Dockerfile.* does NOT run `cargo build`, instead they simply copy cross-built binaries into the container
-    2. Other code: these containers can be .NET or python or whatever else ... the build/Dockerfile.* must do whatever building is required.
-2. Intermediate components: these containers are used as part of the build process and are not used in production explicitly
+In essence, Akri components can be thought of as: 1. Runtime components 1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
+
+{% hint style="info" %}
```
Comment (bfjelds) [inline docs/development/building.md:7]: what does this `{% hint ... %}` look like in an md viewer?  when i click the preview button in github, this file doesn't format well in markdown.

File: docs/development/building.md [hunk 2]
```diff
@@ -1,36 +1,42 @@
...
+
 Building Akri, whether locally or in the automated CI builds, leverages the same set of `make` commands.
 
-In essence, Akri components can be thought of as:
-1. Runtime components
-    1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
-        > Note: For Rust code, build/Dockerfile.* does NOT run `cargo build`, instead they simply copy cross-built binaries into the container
-    2. Other code: these containers can be .NET or python or whatever else ... the build/Dockerfile.* must do whatever building is required.
-2. Intermediate components: these containers are used as part of the build process and are not used in production explicitly
+In essence, Akri components can be thought of as: 1. Runtime components 1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
```
Comment (bfjelds) [inline docs/development/building.md:5]: markdown's automatic list numbering doesn't seem to work when the list is part of a paragraph.

i think the numbers have to be incremented manually or the list needs to be separated as it initially was.

Comment (bfjelds) [reply docs/development/building.md:5]: this seems true for all of the numbered lists in this page.

File: docs/development/building.md [hunk 3]
```diff
@@ -140,9 +148,11 @@ There are two steps to this.  For the sake of this demonstration, only the amd64
     make controller-build-amd64
     # Build the agent container locally for amd64
     make agent-build-amd64
-    ```
-2. Runtime
-    ```bash
+```
+
+1. Runtime
```
Comment (bfjelds) [inline docs/development/building.md:153]: if Build is no longer numbered, Runtime shouldn't be either

File: docs/discovery-handlers/opc-ua.md [hunk 4]
```diff
@@ -1,80 +1,87 @@
...
 
-All of Akri's components can be deployed by specifying values in its Helm chart during an installation. This section will cover the values that should be set to (1) deploy the OPC UA Discovery Handlers and (2) apply a Configuration that tells Akri to discover devices using that Discovery Handler.
+OPC UA \(Object Linking and Embedding for Process Control Unified Architecture\) is a communication protocol for industrial automation. Akri has implemented a Discovery Handler for discovering OPC UA Servers that live at specified endpoints or are registered with specified Local Discovery Servers. Background on the OPC UA Discovery Handler implementation can be found in the [proposal](proposals/opcua.md). To try out using Akri to discover and utilize OPC UA servers, see the [OPC UA end-to-end demo](opcua-demo.md).
```
Comment (shantanoo-desai) [inline docs/discovery-handlers/opc-ua.md:5]: `[proposal]`  link in OPC-UA directs to an non-existing page `https://proposal/opcua.md`. I assume it should be:

    [proposal](../proposals/opcua.md)
    
 similar Linking problems are evident through out the OPC-UA and udev pages.

**Comment by edrickwong** [comment]
> Can you clarify the required changes that you made? And possibly point us to the gitbooks documentation that requests this formating? It would be good to have that content as part of the PR history. ALdo, this is a quite a bit to sift through, so it would help me know where to focus. It looks like it is indentation changes, image name changes, and some content/syntax changes.

The original markdown files under the /docs directory were imported into Gitbook and it generated a tree structure that it can understand, which can be found [here](https://github.com/GitbookIO/gitbook/blob/master/docs/structure.md). When I imported the content, the major changes I did were to make sure Gitbook rendered the content correctly, which required some changes from their CMS and also making sure that most of the links updated. 

There are some image and file names that have been change, which were automatically generated from Gitbook. I can go in and update all of those. None of the code and very little of the actual content was changed.

File: docs/discovery-handlers/opc-ua.md (hunk 4 above)
Comment (edrickwong) [reply docs/discovery-handlers/opc-ua.md:5]: With Gitbook, there's no proposal folder anymore, just a section in the side. I'll update the instructions to reflect this.

**Comment by kate-goldenring** [comment]
Closing due to decision to move Akri documentation to a standalone repository here: https://github.com/deislabs/akri-docs.

**Pull request closed without merging**